pkg/server/http/svc_test.go

```diff
@@ -0,0 +1,1123 @@
+				ID:          "rutherford",
+				Email:       "rutherford@example.com",
+				Displayname: "ErnestRutherFord",
+			},
+			nil,
+		},
+
+		// User with Uid and Gid defined
+		{
+			User{
+				Enabled:     "true",
+				Username:    "thomson",
+				ID:          "thomson",
+				Email:       "thomson@example.com",
+				Displayname: "J. J. Thomson",
+				UIDNumber:   20027,
+				GIDNumber:   30000,
+			},
+			nil,
+		},
+
+		// User with different username and Id
+		{
+			User{
+				Enabled:     "true",
+				Username:    "niels",
+				ID:          "bohr",
+				Email:       "bohr@example.com",
+				Displayname: "Niels Bohr",
+			},
+			nil,
+		},
+
+		// User with special character in username
+		{
+			User{
+				Enabled:     "true",
+				Username:    "schrödinger",
+				ID:          "schrödinger",
+				Email:       "schrödinger@example.com",
+				Displayname: "Erwin Schrödinger",
+			},
+			&Meta{
+				Status:     "error",
+				StatusCode: 400,
+				Message:    "preferred_name 'schrödinger' must be at least the local part of an email",
+			},
+		},
+
+		// User with different userid and email
+		{
+			User{
+				Enabled:     "true",
+				Username:    "planck",
+				ID:          "planck",
+				Email:       "max@example.com",
+				Displayname: "Max Planck",
+			},
+			nil,
+		},
+
+		// User with different userid and email and username
+		{
+			User{
+				Enabled:     "true",
+				Username:    "hisenberg",
+				ID:          "hberg",
+				Email:       "werner@example.com",
+				Displayname: "Werner Hisenberg",
+			},
+			nil,
+		},
+
+		// User without displayname
+		{
+			User{
+				Enabled:  "true",
+				Username: "oppenheimer",
+				ID:       "oppenheimer",
+				Email:    "robert@example.com",
+			},
+			nil,
+		},
+
+		// User wit invalid email
+		{
+			User{
+				Enabled:  "true",
+				Username: "chadwick",
+				ID:       "chadwick",
+				Email:    "not_a_email",
+			},
+			&Meta{
+				Status:     "error",
+				StatusCode: 400,
+				Message:    "mail 'not_a_email' must be a valid email",
+			},
+		},
+
+		// User without email
+		{
+			User{
+				Enabled:  "true",
+				Username: "chadwick",
+				ID:       "chadwick",
+			},
+			&Meta{
+				Status:     "error",
+				StatusCode: 400,
+				Message:    "mail '' must be a valid email",
+			},
+		},
+
+		// User without username
+		{
+			User{
+				Enabled: "true",
+				ID:      "chadwick",
+				Email:   "james@example.com",
+			},
+			&Meta{
+				Status:     "error",
+				StatusCode: 400,
+				Message:    "preferred_name '' must be at least the local part of an email",
+			},
+		},
+
+		// User without userid
+		{
+			User{
+				Enabled:  "true",
+				Username: "chadwick",
+				Email:    "james@example.com",
+			},
+			nil,
+		},
+	}
+
+	for _, ocsVersion := range ocsVersions {
+		for _, format := range formats {
+			for _, data := range testData {
+				formatpart := getFormatString(format)
+				res, err := sendRequest(
+					"POST",
+					fmt.Sprintf("/%v/cloud/users%v", ocsVersion, formatpart),
```

Comment:
IMO it can be reused by using function.

Reply:
Yes, a function that takes the format, ocsVersion and the formatType, so it wont be much different than just using this one line